api/kyverno/v2alpha1/policy_exception_types.go

```diff
@@ -38,28 +42,67 @@ type PolicyException struct {
 	Spec PolicyExceptionSpec `json:"spec"`
 }
 
+// regexVariables represents regex for '{{}}'
+var regexVariables = regexp.MustCompile(`\{\{[^{}]*\}\}`)
+
 // Validate implements programmatic validation
 func (p *PolicyException) Validate() (errs field.ErrorList) {
+	if err := ValidateVariables(p); err != nil {
+		errs = append(errs, field.Forbidden(field.NewPath(""), fmt.Sprintf("Policy Exception \"%s\" should not have variables", p.Name)))
+	}
 	errs = append(errs, p.Spec.Validate(field.NewPath("spec"))...)
 	return errs
 }
 
+func ValidateVariables(polex *PolicyException) error {
+	return objectHasVariables(polex)
+}
+
+func objectHasVariables(object interface{}) error {
```

Comment:
We should move this to utils package and reuse everywhere, there's already an existing one in `objectHasVariables`. This can be improved in a separate PR.